### PR TITLE
Django lesson outline

### PR DESCRIPTION
**Pull request by ashiq-r31**

### CODE REVIEW

File: lessons/django/lesson.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+Performing system checks...
+
+System check identified no issues (0 silenced).
+You have 17 unapplied migration(s). Your project may not work properly until you apply the migrations for app(s): admin, auth, contenttypes, sessions.
+Run 'python manage.py migrate' to apply them.
+May 17, 2019 - 16:09:28
+Django version 2.2.1, using settings 'mysite.settings'
+Starting development server at http://127.0.0.1:8000/
+Quit the server with CONTROL-C.
+```
```
Comment (bennett39) [inline lessons/django/lesson.md:67]: Maybe a quick note about "Don't worry about the _17 unapplied migrations_ warning. We'll learn what that is later..."

Comment (bennett39) [reply lessons/django/lesson.md:67]: Probably should go through these steps again with the most recent version of Django too. The code above references `May 17, 2019`

File: lessons/django/lesson.md [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+```python
+from django.urls import path
+
+from myapp.views import hello
+
+urlpatterns = [
+    # myapp/<str:name>
+    path('<str:name>', hello, name='hello'),
+]
```
Comment (bennett39) [inline lessons/django/lesson.md:240]: Could add this as a second path, beneath the first one above:

```suggestion
urlpatterns = [
    path('', hello, name='hello'),
    path('<str:name>', hello_name, name='hello_name'),
]
```

Then you don't get an error when you visit the index `/` and you can add the code below as a second function (not need to delete/update the existing function)

**Review by bennett39 — COMMENTED** [review]
@ashiq-r31 Few comments, but looks really good overall!

File: lessons/django/lesson.md (hunk 2 above)
Comment (ashiq-r31) [reply lessons/django/lesson.md:240]: would we need to set a default parameter for the first route? think it errors out if the view is expecting one?

File: lessons/django/lesson.md [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+
+-   Create a virtual environment
+-   Create a Django project and app
+-   Serve a custom HTML page!
+
+# Why Django?
+
+-   Known as “the web framework for perfectionists with deadlines,” Django powers some of the biggest websites in the world.
+-   Explain how Django allows us to do full stack development
+-   Distinguish between using Django as a server rendered application and an API server.
```
[the remark this answers is not in the document]
Comment (ashiq-r31) [reply lessons/django/lesson.md:18]: Yeah, perhaps not useful here, but certainly something I'd like to highlight. I'd rather people didn't walk leave the program thinking that we've taught them the default way of building things. 

A lot of people come into the program with pre-existing notions, want to go above and beyond with React, etc. Making the distinction will be valuable before they work on their final projects

File: lessons/django/lesson.md (hunk 2 above)
Comment (bennett39) [reply lessons/django/lesson.md]: @ashiq-r31 Oh I'm suggesting a separate url, view, template, everything

File: lessons/django/lesson.md [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+INSTALLED_APPS = [
+    'myapp.apps.MyappConfig',
+    ... # Leave all the other INSTALLED_APPS
+]
+```
+
+# Migrate the database
+
+Let's migrate those initial models!
+
```
Comment (pab2163) [inline lessons/django/lesson.md:96]:
```suggestion
* Explain what a [migration](https://docs.djangoproject.com/en/3.2/topics/migrations/) is (updating any changes made to the models to be reflected in the database)
* Explain when migrations need to be done
```

**Review by pab2163 — APPROVED** [review]
@ashiq-r31 Looks great! 

I know how you felt about this before, but after seeing this all laid out as one class, I think we'll want to make some videos beforehand. This feels like a lot of content to me, and I suspect that many of the new changes, like using virtual environments and setting up the project/app structure, will either have to take too long or be confusing to people if this is their first time seeing it.

File: lessons/django/lesson.md [hunk 5]
```diff
@@ -0,0 +1,320 @@
+# Before Class
+
+-   Revisit concepts from 'How the web works' content
+-   Install `pyenv`
```
[the remark this answers is not in the document]
Comment (ashiq-r31) [reply lessons/django/lesson.md:4]: @pab2163 I'm thinking of taking `pyenv` and virtual environments stuff altogether. There's tone to cover like you're mentioned and `pyenv` doesn't support Windows.

**Pull request merged**